recipes/default.rb

```diff
@@ -57,7 +57,32 @@
   ports %w(9323)
 end
 
-docker_installation_package 'default'
+node.default['osl-docker']['service']['misc_opts'] = '--live-restore'
+node.default['osl-docker']['service']['host'] = ['unix:///var/run/docker.sock']
+node.default['osl-docker']['service']['host'] << node['osl-docker']['host'] unless node['osl-docker']['host'].nil?
+
+docker_service 'default' do
+  node['osl-docker']['service'].each do |key, value|
+    send(key.to_sym, value)
+  end
+  if node['osl-docker']['tls'] && ::File.exist?('/etc/docker/ssl/key.pem')
+    tls_verify true
+    tls_ca_cert '/etc/docker/ssl/ca.pem'
+    tls_server_cert '/etc/docker/ssl/server.pem'
+    tls_server_key '/etc/docker/ssl/server-key.pem'
+    tls_client_cert '/etc/docker/ssl/cert.pem'
+    tls_client_key '/etc/docker/ssl/key.pem'
+  end
+  if node['osl-docker']['client_only']
+    action [:create, :stop]
+  else
+    if !node['osl-docker']['tls'] || ::File.exist?('/etc/docker/ssl/key.pem')
+      action [:create, :start]
+    else
+      action [:create]
+    end
```

Comment:
I feel like this should be squashed into if elsif else.

Reply:
Done